larpandoracontent/LArMonitoring/EventValidationAlgorithm.cc

```diff
@@ -24,6 +24,7 @@ namespace lar_content
 EventValidationAlgorithm::EventValidationAlgorithm() :
     m_useTrueNeutrinosOnly(false),
     m_testBeamMode(false),
+    m_testBeamHierarchyMode(false),
```

Comment:
How well do you thing this would work if we tried to factorise out the old and new/hierarchy modes into separate algorithms? It would avoid e.g. the re-use of the `*Nu` variables for TB usage, and the Nu vs. TB string name, but will lead to repetition.

Some repetition is OK if it's really straightforward and simple. I wondered about introducing algorithm tools to interpret the final matching info. Could have one tool to print comprehensive matching to terminal, one to print interpreted matching, one to write a tree and one to show things in an event display. Then, instead of setting Boolean flags, simply enable/disable the tools in XML.

Might also need tools to populate the info maps differently for different use cases, or maintain separate algs or the Boolean switch. I'm thinking aloud here and generally liking the idea of getting the event display that matches the matching print out to terminal back in action again!